### DISCUSSION TITLE
Why preferred-width not always works?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3839] [Q&A]
In my app, I wanted to set default width/height of app to bigger value, but using different `preferred-width`/`preferred-height` values not changes default size of window(I used different min-width to show that this property changes, but preferred-width not).

https://github.com/slint-ui/slint/assets/41945903/faca3e7e-e494-443a-8084-5dfa90bc2993

When checking for project for reproduce issue, I found strange situation where HorizontalLayout and ListView in VerticalLayout disabled possibility to change width of item.

Is this a bug?

https://github.com/slint-ui/slint/assets/41945903/5506d848-bf06-4e86-8c9b-9684c556fcf4

**Comment by ogoffart** [comment]
I'm not sure I understand your problem based on these video.
preferred-width/preferred-height  should work, unless the window is restricted further by its minimum or maximum size. This could happen if you have an empty layout whose maximum size is 0.
Note that the constraints will not be applied for 0 exactly because a maximum size of 0 cannot be applied on the windows, I think.
You should add things in your layouts so that their size is not zero.

**Reply by qarmin** [reply]
I already reported this as bug, because it randomly works - https://github.com/slint-ui/slint/issues/3847
